src/jdk.compiler/share/classes/com/sun/tools/javac/comp/AttrContext.java

```diff
@@ -92,6 +92,10 @@ public class AttrContext {
      */
     boolean allowProtectedAccess = false;
 
+    /** Are we attributing a permits clause?
+     */
+    boolean isPermitsClause = false;
```

Comment:
you need to copy the value of this field in the ::dup method as it is done for other fields in this class